modules/images/regenerate-existing-images/background-process/class-perflab-background-process.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * Background process runner class.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Class Perflab_Background_Process.
+ *
+ * Runs the heavy lifting tasks in background in separate process.
+ */
+class Perflab_Background_Process {
+	/**
+	 * Name of the action which will trigger background
+	 * process to run the job.
+	 *
+	 * @const string
+	 */
+	const BG_PROCESS_ACTION = 'background_process_handle_request';
+
+	/**
+	 * Job instance.
+	 *
+	 * @var Perflab_Background_Job
+	 */
+	private $job;
+
+	/**
+	 * Perflab_Background_Process constructor.
+	 *
+	 * Adds the necessary hooks to trigger process handling.
+	 *
+	 * @return void
+	 */
+	public function __construct() {
+		add_action( 'wp_ajax_' . Perflab_Background_Process::BG_PROCESS_ACTION, array( $this, 'handle_request' ) );
+		add_action( 'wp_ajax_nopriv' . Perflab_Background_Process::BG_PROCESS_ACTION, array( $this, 'handle_request' ) );
```

Comment:
For the same file class file, we should use `self::`. It should be updated in other places as well. 
```suggestion
		add_action( 'wp_ajax_' . self::BG_PROCESS_ACTION, array( $this, 'handle_request' ) );
		add_action( 'wp_ajax_nopriv' . self::BG_PROCESS_ACTION, array( $this, 'handle_request' ) );
```